src/backings/structural/container.ts

```diff
@@ -74,18 +77,20 @@ export class ContainerStructuralHandler<T extends ObjectLike> extends Structural
     return Object.entries(this._type.fields).every(([fieldName, fieldType]) => {
       if (fieldType.isBasic()) {
         return fieldType.equals(value1[fieldName], value2[fieldName]);
-      } else {
-        return fieldType.structural.equals(value1[fieldName], value2[fieldName]);
+      }
+      if (fieldType.isComposite()) {
+        return fieldType.structural.equals(value1[fieldName] as ObjectLike, value2[fieldName] as ObjectLike);
       }
     });
   }
   clone(value: T): T {
     const newValue = {} as T;
     Object.entries(this._type.fields).forEach(([fieldName, fieldType]) => {
       if (fieldType.isBasic()) {
-        newValue[fieldName as keyof T] = fieldType.clone(value[fieldName]);
-      } else {
-        newValue[fieldName as keyof T] = fieldType.structural.clone(value[fieldName]);
+        newValue[fieldName as keyof T] = fieldType.clone(value[fieldName]) as T[keyof T];
+      }
+      if (fieldType.isComposite()) {
```

Comment:
why is this needed now?

Reply:
Because typescript is stupid. It cannot (or at least I didn't find a way) to infer is isBasic is false that type is composite. 

That said when going trough code. It was a lot easier to understand code with usage of isComposite to I decided to leave it on. We can make it protected so end users don't see it.